web/ui/src/Hosts/HostsController.js

```diff
@@ -4,33 +4,145 @@
 (function(){
     "use strict";
 
-    controlplane.controller("HostsController", ["$scope", "$routeParams", "$location",
-        "$filter", "resourcesFactory", "authService", "$modalService",
-        "$interval", "$timeout", "$translate", "$notification", "miscUtils", "hostsFactory",
-        "poolsFactory", "servicesFactory", "areUIReady",
-    function($scope, $routeParams, $location, $filter, resourcesFactory,
-    authService, $modalService, $interval, $timeout, $translate, $notification,
-    utils, hostsFactory, poolsFactory, servicesFactory, areUIReady){
-        // Ensure logged in
-        authService.checkLogin($scope);
-
-        $scope.indent = utils.indentClass;
-
-        $scope.resetNewHost = function(){
-            $scope.newHost = {
-                port: $translate.instant('placeholder_port')
+    class HostsController {
+
+        constructor($scope, $routeParams, resourcesFactory, authService,
+                    $modalService, $translate, $notification, areUIReady,
+                    $interval, servicedConfig, log, miscUtils, Host) {
+
+            authService.checkLogin(this);
+
+            this.resourcesFactory = resourcesFactory;
+            this.$modalService = $modalService;
+            this.$translate = $translate;
+            this.$notification = $notification;
+            this.areUIReady = areUIReady;
+            this.$interval = $interval;
+            this.utils = miscUtils;
+            this.params = $routeParams;
```

Comment:
Since there is quite a bit of setting and passing around of these angularjs providers, perhaps they should follow the pattern you used in `Host.js` where there is an intermediate function which receives all of the injected providers and sets them to some module global variables so that all the code can use them. Heres a gist with that change applied to `HostsController.js`: https://gist.github.com/rDr4g0n/359dc29c1d0f38629dd9064472d45ea1